atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbHttpProtocolVersion.java

```diff
@@ -16,16 +16,34 @@
 
 package com.palantir.atlasdb.http;
 
+import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 
 import javax.ws.rs.core.HttpHeaders;
 
+import com.palantir.common.streams.KeyedStream;
 import com.palantir.conjure.java.api.config.service.UserAgent;
 import com.palantir.conjure.java.api.config.service.UserAgents;
 
 public enum AtlasDbHttpProtocolVersion {
-    LEGACY_OR_UNKNOWN,
-    CONJURE_JAVA_RUNTIME;
+    LEGACY_OR_UNKNOWN("1.0"),
+    CONJURE_JAVA_RUNTIME("2.0");
+
+    private static final Map<String, AtlasDbHttpProtocolVersion> KNOWN_VERSION_STRINGS
+            = KeyedStream.of(Arrays.stream(AtlasDbHttpProtocolVersion.values()))
+                    .mapKeys(AtlasDbHttpProtocolVersion::getProtocolVersionString)
+                    .collectToMap();
+
+    private final String protocolVersionString;
+
+    AtlasDbHttpProtocolVersion(String protocolVersionString) {
+        this.protocolVersionString = protocolVersionString;
+    }
+
+    public String getProtocolVersionString() {
+        return protocolVersionString;
+    }
 
     public static AtlasDbHttpProtocolVersion inferFromHttpHeaders(HttpHeaders headers) {
```

Comment:
not a fan of `HttpHeaders` here, do all the request processing inside the filter/exception mapper bla bla, and this should just concern itself with mapping string to `AtlasDbHttpProtocolVersion`

probably should go in the other PR, don't care where